### PR TITLE
Fix Issue #35 - ask for capability size. Too small size could cause errors

### PR DESCRIPTION
**Pull request by gwawiork**

### CODE REVIEW

File: test_conformance/computeinfo/main.c [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "../../test_common/harness/testHarness.h"
 #include "../../test_common/harness/kernelHelpers.h"
 
+
```
Comment (AlexeySachkov) [inline test_conformance/computeinfo/main.c]: Why do you need this empty line?

File: test_conformance/conversions/test_conversions.c [hunk 2]
```diff
@@ -45,6 +45,7 @@
 #include <sys/mman.h>
 #endif
 #include <time.h>
+#include <vector>
```
Comment (AlexeySachkov) [inline test_conformance/conversions/test_conversions.c]: This file has `.c` extensions, but now it includes C++ STL header.
Could you please note this file in #25 or may be it is better to change it to `.cpp` just now?

File: test_conformance/computeinfo/main.c [hunk 3]
```diff
@@ -371,7 +373,12 @@ int getConfigInfo(cl_device_id device, config_info* info)
             size_err = config_size_ret != sizeof(info->config.ull);
             break;
         case type_string:
-            err = clGetDeviceInfo(device, info->opcode, sizeof(info->config.string), &info->config.string, &config_size_ret);
+            err = clGetDeviceInfo(device, info->opcode, 0, NULL, &config_size_set);
+            info->config.string = NULL;
+            if (err == CL_SUCCESS && config_size_set > 0) {
+                info->config.string = (char*)malloc(config_size_set);
+                err = clGetDeviceInfo(device, info->opcode, config_size_set, info->config.string, &config_size_ret);
```
Comment (AlexeySachkov) [inline test_conformance/computeinfo/main.c]: What about `size_err` here?

File: test_conformance/computeinfo/main.c [hunk 4]
```diff
@@ -670,15 +677,20 @@ int getConfigInfos( cl_device_id device )
...
         } else if ( info.opcode == CL_DEVICE_EXTENSIONS ) {
           err = parseExtensions( info.config.string, & extensions );
           if ( err ) {
             total_errors++;
+            free(info.config.string);
             break;
           }
         }
+        if (info.config_type == type_string) {
+            free(info.config.string);
```
Comment (AlexeySachkov) [inline test_conformance/computeinfo/main.c]: Looks like this line should be a little bit less indented (2 extra spaces)

File: test_conformance/computeinfo/main.c [hunk 5]
```diff
@@ -670,15 +677,20 @@ int getConfigInfos( cl_device_id device )
...
           }
         } else if ( info.opcode == CL_DEVICE_EXTENSIONS ) {
           err = parseExtensions( info.config.string, & extensions );
           if ( err ) {
             total_errors++;
+            free(info.config.string);
             break;
           }
         }
+        if (info.config_type == type_string) {
```
Comment (AlexeySachkov) [inline test_conformance/computeinfo/main.c]: What about spaces around `(` and `)` to conform with code style in surrounding code?

**Comment by kpet** [comment]
Commenting on process rather than the change: I'd rather we don't merge new changes before we complete the migration to Github and before we port change from the private Khronos repo to this one.

**Comment by gwawiork** [comment]
I will close this PR and open new with review fixes. I have some issues with this branch while rebasing.

**Pull request closed without merging**